cunumeric/module.py

```diff
@@ -1994,27 +2066,30 @@ def tile(A, reps):
     --------
     Multiple GPUs, Multiple CPUs
     """
-    if not hasattr(reps, "__len__"):
-        reps = (reps,)
+    computed_reps: Union[ndarray, tuple[int, ...]]
+    if isinstance(reps, int):
+        computed_reps = (reps,)
+    else:
+        computed_reps = reps
```

Comment:
Could use some guidance here. I think this change reflects the same intent. But if we need to go back I think several cast will be needs (or perhaps there's another code re-arrangement that will make the types clearer to mypy, I have not thought further on it yet)

Reply:
I think NumPy's `array_like` is anything you can pass to `numpy.array`, which includes `numpy.ndarray` and other sequence-like data structures. I feel we should define our `array_like`, but the type of `compute_reps` seems narrower than it actually should be.

Reply:
> Well Numpy's ArrayLike is quite something:

oooh, I don't think we'd want to go that far... 

> I'd prefer to leave any attempt to introduce a comparable facility to a dedicated PR.

sounds good

> As for the issue at hand, do you think Union[ndarray, Sequence[int]] woudl suffice?

that should work.

Reply:
https://github.com/nv-legate/cunumeric/pull/413/commits/2653a152761cf2809e1c91b785fef0fab2dc8943